ion/marshal.go

```diff
@@ -352,6 +359,31 @@ func (m *Encoder) encodeDecimal(v reflect.Value) error {
 	return m.w.WriteDecimal(d)
 }
 
+func (m *Encoder) encodeWithAnnotation(v reflect.Value) error {
+	original := v
+	fields := fieldsFor(v.Type())
+	for _, field := range fields {
+		if field.annotations {
+			annotations, err := findSubvalue(original, &field)
+			if err != nil {
+				return err
+			}
+			if annotations.Kind() != reflect.Slice {
+				return fmt.Errorf("ion: '%v' is provided for annotations,"+
+					"it must be of type []string", annotations.Kind())
+			}
+			listOfAnnotations := annotations.Interface().([]string)
```

Comment:
You can combine this and a more-accurate version of the above typecheck by using the [two-return-value type assertion](https://tour.golang.org/methods/15):

```go
annos, ok := annotations.Interface().([]string)
if !ok {
  return fmt.Errorf(...)
}
```

Reply:
Thanks for suggesting it. Updated.